jpa/subsystem/src/main/java/org/jboss/as/jpa/service/PersistenceUnitServiceImpl.java

```diff
@@ -270,9 +276,14 @@ public Void run() {
                                         javaNamespaceSetup.teardown(Collections.<String, Object>emptyMap());
                                     }
                                 }
-                                if (proxyBeanManager != null) {
-                                    proxyBeanManager.setDelegate(null);
-                                    proxyBeanManager = null;
+                                final ProxyBeanManager theProxyBeanManager = proxyBeanManager;
```

Comment:
If you're using the double check locking idiom this won't be required.

Reply:
good point